crates/core_arch/src/aarch64/neon.rs

```diff
@@ -232,6 +243,203 @@ extern "C" {
     ) -> int8x16_t;
 }
 
+/// Add pairwise
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(addp))]
+pub unsafe fn vpaddq_u8(a: uint8x16_t, b: uint8x16_t) -> uint8x16_t {
+    vpaddq_u8_(a, b)
+}
+/// Polynomial multiply long
+//poly128_t vmull_p64 (poly64_t a, poly64_t b)
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(pmull))]
+pub unsafe fn vmull_p64(a: poly64_t, b: poly64_t) -> poly128_t {
+    transmute(vmull_p64_(transmute(a), transmute(b)))
+}
+
+/// Macro to template a simd_<something> function that has
+/// the same input and output type
+macro_rules! aarch64_simd_2 {
+    ($name:ident, $type:ty, $simd_fn:ident, $intr:ident) => {
+        #[inline]
+        #[target_feature(enable = "neon")]
+        #[cfg_attr(test, assert_instr($intr))]
+        pub unsafe fn $name(a: $type, b: $type) -> $type {
+            $simd_fn(a, b)
+        }
+    };
+}
+
+/// Macro to template vecq* SIMD intrinsics
+macro_rules! aarch64_simd_ceq {
+    ($name:ident, $type:ty) => {
+        /// Compare bitwise Equal (vector)
+        aarch64_simd_2!($name, $type, simd_eq, cmeq);
+    };
+}
+
+aarch64_simd_ceq!(vceq_s64, int64x1_t);
```

Comment:
@alexcrichton thinking about using macros here... i'm pretty sure this is going to mess stdsimd-verify (it's not going to match that these are functions). 

I don't know of any simple way to improve stdsimd-verify to support this. We would need to do macro-expansion on the syn AST to support this.

Reply:
hmm would generalising all macros to `arm_simd(...)` be a halfway house? it'd be a bit extra to type but if we the stdsimd-verify would only need to look at a singular macro with the types always in one place it might become easier to handle?

Reply:
The simplest way to remain compatible with stdsimd-verify is to provide actual function items for the intrinsics. Anything else is probably going to require macro expansion.

Reply:
If we're having to unroll all the macros it would make sense to do that soonish. An alternative to prevent all the copy&paste would be a build.rs or even shell script to generate an extra file with the corresponding intrinsics but I'm not sure if that'd be 'overoptimizing' for not having to copy and paste a ton of functions around :P. 

I see the macros or some kind of generator be useful for the math functions (add, sub, mul, div) bitwise options (and, or, xor) perhaps (shl, shr) and the compairison operators (eq, ne, lt, le, gt, ge) that does account for a fair bit of functions, since they've not all the same number lets call it 10 (lt was 14) so it's pessimistic about 150 functions that pretty much could be easily generated aside of copy&pasted.

Reply:
Stabilizing any intrinsics requires, among other things, that their type signature be automatically-verified against a spec provided by the vendor, so as long as that's done for any new intrinsic, I'm fine with any approach taken.

Right now, none of the intrinsics added in this PR that are generated behind a macro is verified, so this is something that will need to be resolved before merging this.

Whether you prefer to not use macros, or to add support for expanding macros to stdarch-verify, or to generate the intrinsics via a build.rs, or to have some external script that generate`.rs` files to be included, ... is kind of up to you.

Reply:
Cool! Wanted to make sure that I don't pick an approach that then isn't welcome :D. I'll work something out.

Reply:
Oooookay, here we go, I got rid of all the macros :+1: in favour of generating the intrinsics in `build.rs` Now after I've gone through the whole thing I quite like this approach. It makes it a lot easier to look at and review as it's now written as a kind of specification where code and test are colocated and the whole boilerplate gets generated :).